src/generator/operations.ts

```diff
@@ -451,16 +456,39 @@ function createProtocolRequest(client: string, op: Operation, imports: ImportMan
         text += `\t\t${fieldName} *${bodyParam!.schema.language.go!.name} \`xml:"${tag}"\`\n`;
       }
       text += '\t}\n';
-      body = `wrapper{${fieldName}: &${bodyParam!.language.go!.name}}`;
+      if (!bodyParam!.required) {
+        body = `wrapper{${fieldName}: options.${pascalCase(bodyParam!.language.go!.name)}}`;
+      } else {
+        body = `wrapper{${fieldName}: &${bodyParam!.language.go!.name}}`;
+      }
     } else if (bodyParam!.schema.type === SchemaType.DateTime && (<DateTimeSchema>bodyParam!.schema).format === 'date-time-rfc1123') {
       // wrap the body in the custom RFC1123 type
-      text += `\taux := ${bodyParam!.schema.language.go!.internalTimeType}(${body})\n`;
+      if (!bodyParam!.required) {
+        text += `\taux := ${bodyParam!.schema.language.go!.internalTimeType}(options.${pascalCase(body)})\n`;
+      } else {
+        text += `\taux := ${bodyParam!.schema.language.go!.internalTimeType}(${body})\n`;
+      }
       body = 'aux';
+    } else {
+      // final check for each body param to know if it is optional
+      if (!bodyParam!.required) {
```

Comment:
Nit: Can be replaced with `setOptionsPrefix = !bodyParam!.required` to eliminate nested `if` block.